spacy/tokens/doc.pyx

```diff
@@ -1610,22 +1610,20 @@ cdef class Doc:
 
         if doc_json.get("underscore_token", {}):
             for token_attr in doc_json["underscore_token"]:
-                token_start = doc_json["underscore_token"][token_attr]["token_start"]
-                value = doc_json["underscore_token"][token_attr]["value"]
-
                 if not Token.has_extension(token_attr):
                     Token.set_extension(token_attr)
-                self[token_start]._.set(token_attr, value)
+                for token in doc_json["underscore_token"][token_attr]:
```

Comment:
In the code base in general calling a non-`Token` variable `token` can get kind of confusing. Maybe `token_entry`? `token_dict`? `token_ext_data`? ???

```suggestion
                for token_ext_data in doc_json["underscore_token"][token_attr]:
```